### PR TITLE
Retrospective on the May-July 2024 roadmap

### PR DESCRIPTION
**Pull request by DJMcNab**
[RENDERED](https://github.com/linebender/linebender.github.io/blob/roadmap-retro/content/blog/2024-08-23-roadmap-may-2024-retro.md)

Current status: This is complete besides the harfruzz section, and is awaiting review

### CODE REVIEW

File: content/blog/2024-08-23-roadmap-may-2024-retro.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+The [stroke expansion paper](https://linebender.org/gpu-stroke-expansion-paper/) documents a key part of Vello's pipeline.
+This was presented at ACM High Performance Graphics (HPG) 2024, placing 3rd Place in the Best Paper Award.
+
+## Masonry [(original)](@/blog/2024-06-17-roadmap-may-2024.md#masonry)
+
+### Complete text input [(original)](@/blog/2024-06-17-roadmap-may-2024.md#complete-text-input)
+
+- **Priority**: Highest
+- **Status**: Not that much progress.
+  There were some community contributions such as
```
Comment (matta) [inline content/blog/2024-08-23-roadmap-may-2024-retro.md:60]: incomplete sentence?

File: content/blog/2024-08-23-roadmap-may-2024-retro.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+### Better tracing for debugging [(original)](@/blog/2024-06-17-roadmap-may-2024.md#better-tracing-for-debugging)
+
+- **Priority**: High
+- **Status**: [xilem#384](https://github.com/linebender/xilem/pull/384) implements most of this, the rest is deferred.
+  There are known issues with the size of the log files created by default in very large apps.
+
+### Fix glaring documentation problems [(original)](@/blog/2024-06-17-roadmap-may-2024.md#fix-glaring-documentation-problems)
+
+- **Priority**: High
+- **Status**: The issues mentioned in this item were addressed in [xilem#329](https://github.com/linebender/xilem/pull/329), and.
```
Comment (matta) [inline content/blog/2024-08-23-roadmap-may-2024-retro.md:71]: incomplete sentence?

File: content/blog/2024-08-23-roadmap-may-2024-retro.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+### Write “widgets in arena” RFC [(original)](@/blog/2024-06-17-roadmap-may-2024.md#write-widgets-in-arena-rfc)
+
+- **Priority**: High
+- **Status**: The RFC was [rfcs#6](https://github.com/linebender/rfcs/pull/6), which was mostly implemented in [xilem#396](https://github.com/linebender/xilem/pull/396).
+
+### Write “pass order” RFC [(original)](@/blog/2024-06-17-roadmap-may-2024.md#write-pass-order-rfc)
+
+- **Prioriy**: High
+- **Status**: [rfcs#7](https://github.com/linebender/rfcs/pull/7) is in draft, and several important parts of the RFC have been completed.
+  We have also learned several important lessons about the RFC process, including.
```
Comment (matta) [inline content/blog/2024-08-23-roadmap-may-2024-retro.md:89]: incomplete sentence?

**Review by matta — COMMENTED** [review]
FYI the "rendered" link in the first comment is outdated. I noticed because the conclusion had a sentence fragment that is not present in the current state of the PR. It should be https://github.com/linebender/linebender.github.io/blob/28239f4f66851e667ccde05ca51176fca64157a1/content/blog/2024-08-23-roadmap-may-2024-retro.md

File: content/blog/2024-08-23-roadmap-may-2024-retro.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+- **Priority**: Low
+- **Status**: This was not completed in the form envisioned.
+  However, Nico contributed two examples, showing how to use Parley with Swash ([parley#54](https://github.com/linebender/parley/pull/54)) and tiny-skia ([parley#55](https://github.com/linebender/parley/pull/55)).
+  We have decided not to carry this forward into the next roadmap.
+
+### Progress beyond roadmap {#parley-extras}
+
+[parley#56](https://github.com/linebender/parley/pull/56) added support for Emoji, which is used with the [Vello PRs](#vello-extras).
+
+<!-- TODO: Accurate sentence on Harfruzz -->
```
Comment (raphlinus) [inline content/blog/2024-08-23-roadmap-may-2024-retro.md:150]: Here's my proposed sentence:

Recently the [harfruzz] project, a fork of rustybuzz to be integrated fully with fontations, has emerged as an extremely promising direction for text shaping for the Rust ecosystem. We plan to use it as the basis for text layout.

**Comment by DJMcNab** [comment]
Thanks for the reviews, applied the fixes.

It's not entirely clear to me how that RENDERED link got into the state it was previous. Fwiw, linking to a specific commit in that kind of RENDERED link is never the right behaviour because it will not be updated properly. Thanks for pointing out I had inadvertently used that format!

**Comment by matta** [comment]
> It's not entirely clear to me how that RENDERED link got into the state it was previous. Fwiw, linking to a specific commit in that kind of RENDERED link is never the right behaviour because it will not be updated properly. Thanks for pointing out I had inadvertently used that format!

Makes sense. Side question: is there a place in the GitHub UI where the rendered link is available? All I've found is navigating to "Files Changed" -> https://github.com/linebender/linebender.github.io/pull/59/files -> hamburger menu on the `.md` file in question -> select the "View File" menu item. But that gives me a "permalink" style URL, tied to a commit.

**Comment by DJMcNab** [comment]
Ah, that is the process I went through to get the original URL, that explains it. The way I've found to be correct is to click on the branch name (under the PR title), then navigate to the correct file from there. But it's not ideal

File: content/blog/2024-08-23-roadmap-may-2024-retro.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+- **Status**: Basic translation of touch events was added in ([xilem#313](https://github.com/linebender/xilem/pull/313)).
+  This unblocked using Masonry on devices where pointer input is not expected (primarily on Android).
+  We have split the remainder of this task into two tasks, with a priority placed on completing the parts which unblock other work (such as scrolling on Android).
+
+### Progress beyond roadmap {#masonry-extras}
+
+Masonry is in a state of flux, however is starting to see a few exploratory users.
+To unblock these users, some additional important features were added outside of the roadmap, such as:
+
+- Support for embedding Masonry inside a pre-existing event loop ([xilem#417](https://github.com/linebender/xilem/pull/417)).
```
Comment (PoignardAzur) [inline content/blog/2024-08-23-roadmap-may-2024-retro.md:113]: The link is a 404. I *think* you're referring to [xilem#325](https://github.com/linebender/xilem/pull/325)?

File: content/blog/2024-08-23-roadmap-may-2024-retro.md [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+Xilem has also seen some very useful changes not tracked above, including many community contributions.
+Some highlights of these contributions are:
+
+- Philipp Mildenberger contributed heavily to Xilem's core design, such as by restoring the ability to return `ViewSequence`s by returning to a previous workaround for Rust's overly-cautious trait solver ([xilem#472](https://github.com/linebender/xilem/pull/394)).
+  He also extended the Xilem API by restoring `OneOf` ([xilem#394](https://github.com/linebender/xilem/pull/394)), and implementing various `Adapt` nodes ([xilem#401](https://github.com/linebender/xilem/pull/401)).
+  Additionally, he picked up the work to expand `Flex` to use the [new Xilem Core](#rework-core) in [xilem#428](https://github.com/linebender/xilem/pull/428).
+- Jared O'Connell contributed two new examples, a calculator ([xilem#467](https://github.com/linebender/xilem/pull/467)), and a stopwatch ([xilem#492](https://github.com/linebender/xilem/pull/492)).
+
+There has also been lots of really good progress on Xilem Web, but we do not roadmap Xilem Web and so is not discussed here.
```
Comment (PoignardAzur) [inline content/blog/2024-08-23-roadmap-may-2024-retro.md:208]:
```suggestion
There has also been lots of really good progress on Xilem Web, but we do not roadmap Xilem Web and so we do not cover that progress here.
```

**Review by PoignardAzur — APPROVED** [review]
LGTM.

File: content/blog/2024-08-23-roadmap-may-2024-retro.md (hunk 5 above)
Comment (DJMcNab) [reply content/blog/2024-08-23-roadmap-may-2024-retro.md:113]: No, that link is correct

GitHub has deleted the pull request, but the change I'm referring to happened in that PR.

Comment (DJMcNab) [reply content/blog/2024-08-23-roadmap-may-2024-retro.md]: I've linked to https://github.com/linebender/xilem/commit/cf3530097baf0fc5c458593e215a19973572b991 instead.

**Comment by PoignardAzur** [comment]
New conclusion is good.

**Pull request merged**